rust/core-lib/src/editor.rs

```diff
@@ -961,6 +961,20 @@ impl Editor {
         self.view.collapse_selections(&self.text);
     }
 
+    fn transform_text(&mut self, transform_function: &Fn(String) -> String) {
+        let mut builder = delta::Builder::new(self.text.len());
+
+        for region in self.view.sel_regions() {
```

Comment:
There's a sort of discussion question, here: what is the granularity of this operation? On the mac, the `uppercaseWord` / `lowercaseWord` methods are supposed to transform any word that overlaps with the selection. What is our desired behaviour with an empty selection region (a caret)? On macOS, if you map a key to the `uppercaseWord:` selector, then (in TextEdit, say) it expands your selection to the full word before uppercasing. In sublime text, uppercase / lowercase commands operate on the selected region, or on the containing word in the caret case.

My personal feeling here is that in the caret case, we should be applying the operation to the containing word. Does that seem reasonable?

Reply:
I concur that it should only do it for the selection within core.  For the caret case I think it shouldn't do anything.  It would be the front-end's job I think to go with whatever makes sense for the platform.  For OSX if we want TextEdit-like behaviour the front-end should adjust the selection(s) to word boundaries.  If you wanted to adjust the words without modifying the selection, the frontend could adjust to word boundaries, make transformation, adjust back.